src/git.ts

```diff
@@ -143,23 +142,46 @@ export function parseGitDiffOutput(output: string): File[] {
   return files
 }
 
-export async function listAllFilesAsAdded(): Promise<File[]> {
-  core.startGroup('Listing all files tracked by git')
-  let output = ''
-  try {
-    output = (await exec('git', ['ls-files', '-z'])).stdout
-  } finally {
-    fixStdOutNullTermination()
-    core.endGroup()
-  }
+function mergeStatusNumstat(statusEntries: FileStatus[], numstatEntries: FileNumstat[]): File[] {
+  const statusMap: {[key: string]: FileStatus} = {}
+  statusEntries.forEach(f => statusMap[f.filename] = f)
 
-  return output
-    .split('\u0000')
-    .filter(s => s.length > 0)
-    .map(path => ({
-      status: ChangeStatus.Added,
-      filename: path
-    }))
+  return numstatEntries.map(f => {
+    const status = statusMap[f.filename]
+    if (!status) {
+      throw new Error(`Cannot find the status entry for file: ${f.filename}`);
+    }
+    return {...f, status: status.status}
+  })
+}
+
+export async function getGitDiffStatusNumstat(diffArg: string) {
+  const statusFiles = await gitDiffNameStatus(diffArg).then(parseGitDiffNameStatusOutput)
+  const numstatFiles = await gitDiffNumstat(diffArg).then(parseGitDiffNumstatOutput)
+  return mergeStatusNumstat(statusFiles, numstatFiles)
+}
+
+export function parseGitDiffNumstatOutput(output: string): FileNumstat[] {
+  const rows = output.split('\u0000').filter(s => s.length > 0)
+  return rows.map(row => {
+    const tokens = row.split('\t')
+    // For the binary files set the numbers to zero. This matches the response of Github API.
+    const additions = tokens[0] == '-' ? 0 : Number.parseInt(tokens[0])
+    const deletions = tokens[1] == '-' ? 0 : Number.parseInt(tokens[1])
+    return {
+      filename: tokens[2],
+      additions,
+      deletions,
+    }
+  })
+}
+
+
+export async function listAllFilesAsAdded(): Promise<File[]> {
+  return core.group(`Listing all files tracked by git`, async () => {
+    const emptyTreeHash = (await exec('git', ['hash-object', '-t', 'tree', '/dev/null'])).stdout
+    return getGitDiffStatusNumstat(emptyTreeHash)
```

Comment:
The `git ls-files` does not return the data about the number of lines added. Comparing the current revision to an empty tree both allows getting the stat data, and lets us reuse the existing diff logic.